Mathlib/Tactic/Linter/Lint.lean

```diff
@@ -175,6 +175,8 @@ def getLinterHash (o : Options) : Bool := Linter.getLinterValue linter.endOf o
 def endOfLinter : Linter where run := withSetOptionIn fun stx ↦ do
     -- Only run this linter at the end of a module.
     unless stx.isOfKind ``Lean.Parser.Command.eoi do return
+    unless #[`Mathlib, `test, `Archive, `Counterexamples].contains (← getMainModule).getRoot do
+      return
```

Comment:
I think it would be good to reference https://github.com/leanprover/lean4/pull/3403 here, as a breadcrumb for providing options to control this when they (soon!) become available.